OptimoveSDK/Sources/Classes/OptimoveConfig.swift

```diff
@@ -299,18 +343,53 @@ open class OptimoveConfigBuilder: NSObject {
             }
             Logger.info("\(OptimobileConfig.self) building skipped.")
             return nil
-        }()
+        }()    
+
+        let preferenceCenterConfig: PreferenceCenterConfig? = getPreferenceCenterConfig()
 
         return OptimoveConfig(
             features: features,
             tenantInfo: tenantInfo,
-            optimobileConfig: optimobileConfig
+            optimobileConfig: optimobileConfig,
+            preferenceCenterConfig: preferenceCenterConfig
         )
     }
-}
+
+    private func getPreferenceCenterConfig() -> PreferenceCenterConfig? {
+            //I see Android throws, but here we don't (I don't see throw in the build methods at all), should I be adding it? 
+            guard features.contains(.preferenceCenter) else {
+                Logger.error("Cannot set credentials for preference center as it is not in the desired feature set")
+                return nil
+            }
+
+            guard let credentials = self.preferenceCenterCredentials else {
+                Logger.error("Could not find preference center credentials")
+                return nil
+            }
+
+            do {
+                let args = try PreferenceCenterArguments(base64: credentials)
+
+                return PreferenceCenterConfig(
+                    region: args.region,
+                    tenantId: args.tenantId,
+                    brandGroupId: args.brandGroupId
+                )
+
+            } catch {
+                //I see Android throws, but here we don't (I don't see throw in the build methods at all), should I be adding it?
```

Comment:
to me it makes more sense throwing -- like if they provide garbage string, better to throw than silently not init PC

Reply:
@eligutovsky thoughts on this? Will most likely add the `throws` and handle, but wondering if there was a reason behind this? (build method in `OptimoveConfig` not throwing)

Reply:
i think the reason was smth along "we want to do everything not to crash the app" . i guess both ways possible

in android and previously in all mobile sdks the idea was that if they give bad credentials -- our sdk cannot function correctly, it could create more inquiries, be more obscure to customers. Since they provide credentials during initialization, if they do any app testing at all -- it would be very hard for them to miss the crash / release app with bad credentials.

After delayed init was added, guess it is more possible if credential strings can come from somewhere dynamically. If this actually happened and someone released app which could load bad PC credentials -- would we like to crash the app for them? Probably not.

so, mind change :) let's keep ios sdk consistent within itself and never crash if they give bad strings . Android will check, may also adjust for PC only. Or may do a field test -- which one works better :)

Reply:
but if this actually happened in testing -- crashing would be better. 

mb behaviour could even be 

prod -- dont crash, dev -- crash
delayed -- dont crash, normal init -- crash

So, the answer kinda depends on how well they test their apps / how much more time we want to spend on crash dont crash.

since i think not much, let's not crash on ios